support/util/util.go

```diff
@@ -285,3 +287,69 @@ func FirstUsableIP(cidr string) (string, error) {
 	ip[len(ipNet.IP)-1]++
 	return ip.String(), nil
 }
+
+// PredicatesForHostedClusterAnnotationScoping returns predicate filters for all event types that will ignore incoming
+// event requests for hostedcluster resources that do not match the "scope" annotation
+// specified in the HOSTEDCLUSTERS_SCOPE_ANNOTATION env var.  If not defined or empty, the default behavior is to accept all events for hostedclusters that do not have the annotation.
+// The ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING env var must also be set to "true" to enable the scoping feature.
+func PredicatesForHostedClusterAnnotationScoping() predicate.Predicate {
+	enableHostedClusterAnnotationScopingEnv := os.Getenv("ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING")
```

Comment:
Do we need a separate env var to enable scoping? Wouldn't the presence of a value for HOSTEDCLUSTERS_SCOPE_ANNOTATION imply that you're enabling it?

Reply:
Having a dedicated env var `ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING` for toggling on/off the scoping feature allows a "normal" single operator setup to run EXACTLY as it does today by not specifying any new env var.  Trying to use just the presence of `HOSTEDCLUSTERS_SCOPE_ANNOTATION` would require running through the new "get hostedcluster" logic within `PredicatesForHostedClusterChildResourcesAnnotationScoping` for every incoming event to compare annotations.  This logic would still be required even if the `HOSTEDCLUSTERS_SCOPE_ANNOTATION` env var is empty, since by not specifying it we would expect that operator to ignore events for any hostedclusters that _are_ scoped.  So by having the separate env var to enable the feature, we can skip that little bit of extra processing and preserve code efficiency in the base path for the majority of operator use cases.

---

Example dual operator env configurations where one operator serves "production" clusters while the other serves "test" clusters:

**Operator 1 (production):**
`ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING` = "true"
`HOSTEDCLUSTERS_SCOPE_ANNOTATION` (NOT DEFINED)

Behavior:
Only process events for hostedcluster resources that DO NOT have the `hypershift.openshift.io/scope` annotation defined.


**Operator 2 (test):**
`ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING` = "true"
`HOSTEDCLUSTERS_SCOPE_ANNOTATION` = "test"

Behavior:
Only process events for hostedcluster resources that DO have the `hypershift.openshift.io/scope` annotation defined with a value of "test".